test/unit/multiplication/test_multiplication_triangular.cpp

```diff
@@ -97,6 +108,52 @@ void testTriangularMultiplication(blas::Side side, blas::Uplo uplo, blas::Op op,
                     40 * (mat_bh.size().rows() + 1) * TypeUtilities<T>::error);
 }
 
+template <class T, Backend B, Device D>
+void testTriangularMultiplication(comm::CommunicatorGrid grid, blas::Side side, blas::Uplo uplo,
+                                  blas::Op op, blas::Diag diag, T alpha, SizeType m, SizeType n,
+                                  SizeType mb, SizeType nb) {
+  std::function<T(const GlobalElementIndex&)> el_op_a, el_b, res_b;
+
+  LocalElementSize size_a(m, m);
+  TileElementSize block_size_a(mb, mb);
+
+  if (side == blas::Side::Right) {
+    size_a = {n, n};
+    block_size_a = {nb, nb};
+  }
+
+  Index2D src_rank_index(std::max(0, grid.size().rows() - 1), std::min(1, grid.size().cols() - 1));
+  GlobalElementSize sz_a = globalTestSize(size_a);
+  Distribution distr_a(sz_a, block_size_a, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_ah(std::move(distr_a));
+
+  LocalElementSize size_b(m, n);
+  TileElementSize block_size_b(mb, nb);
+  GlobalElementSize sz_b = globalTestSize(size_b);
+  Distribution distr_b(sz_b, block_size_b, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_bh(std::move(distr_b));
+
+  if (side == blas::Side::Left)
+    std::tie(el_op_a, res_b, el_b) =
+        getLeftTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, static_cast<T>(1.0) / alpha, m);
+  else
+    std::tie(el_op_a, res_b, el_b) =
+        getRightTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, static_cast<T>(1.0) / alpha, n);
```

Comment:
```suggestion
  auto [el_op_a, res_b, el_b] = []{
      if (side == blas::Side::Left)
          return getLeftTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, static_cast<T>(1.0) / alpha, m);
      else
          return getRightTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, static_cast<T>(1.0) / alpha, n);
    }();
```
and then removing the declaration at the top of the function. But as I said in the other comment, it's arguable whether this is actually better...

Reply:
We have used structured bindings throughout the test (thank you @msimberg for help!). What do you think about this approach @rasolca @teonnik @albestro?

Reply:
I think that everything can be simplified if we provide a `getTriangularSystem(side, uplo, op, ...)` function.
There are 2 ways to do it:
- either combine the two functions in a single one
- add a dispatcher function

Reply:
I prefer the approach without structured bindings in this case, wrapping it in a lambda just to use structured bindings is a bit of a hack. Either of the two approaches suggested by Raffaele seem good to me.

Reply:
I've added something similar to a dispatcher function (https://github.com/eth-cscs/DLA-Future/blob/027558d0eae7c15b0e579ee4f5eed20eaff3a893/test/unit/multiplication/test_multiplication_triangular.cpp#L73-L83) for the first two cases (`get...TriangularSystem`), but then I use it in combination with structured bindings actually. 
On the other hand, I've used structured binginds in the nested loops (like here https://github.com/eth-cscs/DLA-Future/blob/027558d0eae7c15b0e579ee4f5eed20eaff3a893/test/unit/multiplication/test_multiplication_triangular.cpp#L163). 
What do you think about it?